toolboxTreppe/toolboxTreppe.ino

```diff
@@ -93,23 +93,22 @@ void animation1() {
       ++m_x;
       xA = 5 - m_x;
       xB = 5 + m_x;
-      if (m_x < 0) {
-        m_x = 0;
-      }else if (m_x > 5) {
+      if (m_x > 5) {
         m_x = 0;
         ++m_y;
         //hue++;
         fadeToBlackBy(leds, NUM_LEDS, 20);
-        if (m_y > kMatrixHeight-1 || m_y < 0) {
+        if (m_y > kMatrixHeight-1) {
           m_y = 0;
         }
       }
     }
     if (currentMillis - onewayMillis > onewayTime) {
       onewayMillis = currentMillis;
-      --gegenstrecke;
-      if (gegenstrecke < 0 || gegenstrecke > kMatrixHeight-1) {
+      if (gegenstrecke == 0) {
         gegenstrecke = kMatrixHeight-1;
+      } else {
+        --gegenstrecke;
```

Comment:
hier würd ich auch nochmal hingucken

Reply:
das ist in der Nachricht darüber erklärt 
nachdem gegenstrecke jetzt auch eine uint8_t ist muss das auch sorum erfolgen, da sie ja nicht mehr negativ werden kann (könnte natürlich auch auf 255 gecheckt werden) 
aber man spart sich so in dem Durchlauf in dem auf kmatrixheight -1 gesetzt wird das --gegenstrecke